controllers/routingctrl/reconcile_resources.go

```diff
@@ -124,24 +124,26 @@ func (r *Controller) ensureResourceHasFinalizer(ctx context.Context, target *uns
 	return nil
 }
 
+// extractExportModes retrieves the enabled export modes from the target's annotations.
 func (r *Controller) extractExportModes(target *unstructured.Unstructured) []spi.RouteType {
-	exportModes, exportModeFound := target.GetAnnotations()[annotations.RoutingExportMode("").Key()]
-	if !exportModeFound {
+	targetAnnotations := target.GetAnnotations()
+	if targetAnnotations == nil {
 		return nil
 	}
 
-	exportModesSplit := strings.Split(exportModes, ";")
-	validRouteTypes := make([]spi.RouteType, 0, len(exportModesSplit))
-
-	for _, exportMode := range exportModesSplit {
-		routeType := spi.RouteType(strings.TrimSpace(exportMode))
-		if spi.IsValidRouteType(routeType) {
-			validRouteTypes = append(validRouteTypes, routeType)
-		} else {
-			r.log.Info("Invalid route type found",
-				"invalidRouteType", routeType,
-				"resourceName", target.GetName(),
-				"resourceNamespace", target.GetNamespace())
+	validRouteTypes := make([]spi.RouteType, 0)
+
+	for key, value := range targetAnnotations {
+		if strings.HasPrefix(key, annotations.RoutingExportModePrefix) && value == "true" {
+			routeType := spi.RouteType(strings.TrimPrefix(key, annotations.RoutingExportModePrefix))
+			if spi.IsValidRouteType(routeType) {
+				validRouteTypes = append(validRouteTypes, routeType)
+			} else {
```

Comment:
How about encapsulating the trim logic in `IsValudRouteType` itself?

Also PTAL at the latest #70 changes as this func might need to be moved to routing pkg too.

Reply:
[f0e3664](https://github.com/maistra/odh-platform/pull/69/commits/f0e3664706aec33f5b8c60dea8e9ab1130050918) encapsulated logic here, will move in future commit

Reply:
[55ec934](https://github.com/maistra/odh-platform/pull/69/commits/55ec93450eb6430a624b1030b827f9478736b603)

 moved IsValidRouteType into `routing`